RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitProducer.cc

```diff
@@ -31,6 +42,48 @@ EcalUncalibRecHitProducer::~EcalUncalibRecHitProducer()
         delete worker_;
 }
 
+void EcalUncalibRecHitProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+
+  EcalUncalibRecHitFillDescriptionWorkerFactory* factory = EcalUncalibRecHitFillDescriptionWorkerFactory::get(); 
+  std::vector<edmplugin::PluginInfo> infos = factory->available();
+ 
+  {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("EBdigiCollection", edm::InputTag("ecalDigis","ebDigis"));
+    desc.add<std::string>("EEhitCollection", "EcalUncalibRecHitsEE");
+    desc.add<edm::InputTag>("EEdigiCollection", edm::InputTag("ecalDigis","eeDigis"));
+    desc.add<std::string>("EBhitCollection", "EcalUncalibRecHitsEB");
+
+    auto itInfos = infos.begin();
+    assert(itInfos != infos.end());
+
+    std::auto_ptr<edm::ParameterDescriptionCases<std::string>> s;
+    {
+      s = itInfos->name_ >> edm::ParameterDescription<edm::ParameterSetDescription>("algoPSet", EcalUncalibRecHitFillDescriptionWorkerFactory::get()->create(itInfos->name_)->getAlgoDescription(), true);
+    }
+    for (++itInfos; itInfos != infos.end(); ++itInfos)
```

Comment:
It would be helpful if a comment was added here explaining why this separate block ({}) is necessary.